katdal/datasources.py

```diff
@@ -263,6 +263,36 @@ def _ensure_prefix_is_set(chunk_info, telstate):
     return chunk_info
 
 
+def _upgrade_chunk_info(chunk_info, improved_chunk_info):
+    """Replace chunk info items with better ones while preserving number of dumps."""
+    for key, improved_info in improved_chunk_info.items():
+        original_info = chunk_info.get(key, improved_info)
+        n_original_dumps = original_info['shape'][0]
+        chunks = improved_info['chunks']
+        time_chunks = chunks[0]
+        n_improved_dumps = sum(time_chunks)
```

Comment:
Looks a bit odd that n_original_dumps is taken from shape but n_improved_dumps is taken from summing chunks.

Reply:
I had `n_improved_dumps = improved_info['shape'][0]` at first, then I removed the line completely, just doing `sum(time_chunks)` wherever needed. Then I relented and added the line back in. I'll go back to the original then - it seems better now.